packages/nitro-protocol/contracts/AssetHolder.sol

```diff
@@ -35,23 +36,23 @@ contract AssetHolder is IAssetHolder {
         uint256 finalPayoutAmount;
         uint256 firstNewAllocationItemAmount;
 
-        for (uint256 i = 0; i < allocation.length; i++) {
+        for (uint256 i = 0; i < allocation.length; i = i.add(1)) {
```

Comment:
So the reasoning is that we know that allocation.length cannot exceed the maximum integer, because allocation was passed in as calldata, and an array that long in calldata would cost too much gas?